python-package/lightgbm/sklearn.py

```diff
@@ -330,7 +330,8 @@ def fit(self, X, y,
             sample_weight=None, init_score=None, group=None,
             eval_set=None, eval_names=None, eval_sample_weight=None,
             eval_class_weight=None, eval_init_score=None, eval_group=None,
-            eval_metric=None, early_stopping_rounds=None, verbose=True,
+            eval_metric=None, eval_train_metric=None, eval_valid_metric=None,
```

Comment:
I don't think that these new params are really needed. One can achieve the needed result in their rare case with `params`.

Reply:
>  I don't think that these new args are really needed. 
> One can achieve the needed result in their rare case with params.

I don't think they are not rere cases, because CatBoost, which is one of the most famous GBDT library and was acepted in NIPS 2018, supports setting of individual metrics to training and validation.
This PR (change) allows us to set evaluation metric different from objective function, which is useful and needed and thus CatBoost implemetned it.

cf. `eval_metric` in
https://tech.yandex.com/catboost/doc/dg/concepts/python-reference_parameters-list-docpage/#python-reference_parameters-list.

However, as you you pointed, `eval_metric`, `eval_train_metric` and `eval_valid_metric` are confusing for many users...

Reply:
@guolinke What do you think? My position is that corresponding keys in `params` is enough because new arguments introduce huge number of possible combination which is very hard to support.

I remember we were getting rid of some arguments in favor of their aliases in `params` some time ago (`max_bin` argument is the first thing that comes to my mind). So, new args is against this policy.